apps/researcher/src/app/[locale]/communities/community-card.tsx

```diff
@@ -32,6 +34,27 @@ async function MembershipCount({communityId, locale}: MembershipCountProps) {
   });
 }
 
+interface MembershipCountProps {
+  communityId: string;
+  locale: string;
+}
+
+async function ObjectListCount({communityId, locale}: MembershipCountProps) {
+  const t = await getTranslator(locale, 'Communities');
+
+  let objectLists = [];
+  try {
+    objectLists = await objectList.getByCommunityId(communityId);
+  } catch (err) {
+    console.error(err);
+    return t('objectListCountError');
+  }
+
+  return t.rich('objectListCount', {
+    count: objectLists.length,
```

Comment:
Does Clerk have a way of counting the records? This implementation works fine, but for this functionality it looks a bit 'heavy': we retrieve all the records but only need the number of records?

Reply:
Good point. This `ObjectListCount` is not Clerk data but from our database. But this is still heavy for our database, so I have added an `objectList.countByCommunityId` function with a SQL `count` function. See commit https://github.com/colonial-heritage/colonial-collections/pull/248/commits/4a53e3443461c4377f0cb7c8eecf3bd526a1c39a

I have also added a `Suspense` around this component so the page can load without all these calls being made.

Above this component, there is also the `MembershipCount` component with a call for all members just for a count of the numbers. This is also heavy; I do this call because the `includeMembersCount` of `clerkClient.organizations.getOrganizationList` does not work. I have asked about this in Discord, and they responded that the fix is in the next release. When this is added, it will be one call per community less.

The caching will also help keep this page not too heavy.